src/server.rs

```diff
@@ -284,6 +283,31 @@ pub struct SendResponse<B: IntoBuf> {
     inner: proto::StreamRef<B::Buf>,
 }
 
+/// Send a response to a promised request
+///
+/// A `SendPushedResponse` instance is provided when promising a request and is used
+/// to send the associated response to the client. It is also used to
+/// explicitly reset the stream with a custom reason.
+///
+/// It can not be used to initiate push promises.
+///
+/// If the `SendPushedResponse` instance is dropped without sending a response, then
+/// the HTTP/2.0 stream will be reset.
+///
+/// See [module] level docs for more details.
+///
+/// [module]: index.html
+pub struct SendPushedResponse<B: IntoBuf> {
+    inner: SendResponse<B>,
+}
+
+impl<B: IntoBuf + fmt::Debug> fmt::Debug for SendPushedResponse<B>
```

Comment:
If there's a reason this can't be derived, a comment explaining that would be good...

Reply:
Yep, and I'm really not sure that's the best way to solve the issue.
It fails because `#[derive(Debug)]` on `SendResponse` requires `<B as bytes::IntoBuf>::Buf: fmt::Debug`

I _think_ this is because of https://github.com/rust-lang/rust/issues/26925

Actually `#[derive(Debug)]` introduces these too strict bounds from `SendResponse` through `StreamRef`, `SendBuffer`, `Buffer` and `Slab`

Another option would be to include the `fmt::Debug` bound on the `SendPushedResponse` type itself.

Any thoughts?

Reply:
In that case, I think the manual impl is correct. Let's leave it as-is, and add a comment stating why it can't be derived so that this is documented for future readers.